tensorflow/lite/experimental/shlo/ops/dot_general.cc

```diff
@@ -0,0 +1,441 @@
+                          Tensor& output) {
+  using StorageT = StorageType<storage_type>;
+  const StorageT* lhs_data = lhs.GetDataAs<storage_type>();
+  const StorageT* rhs_data = rhs.GetDataAs<storage_type>();
+  StorageT* output_data = output.GetDataAs<storage_type>();
+  const size_t lhs_size = lhs.NumElements();
+  const DimensionSize rhs_size = rhs.NumElements();
+  const DimensionSize output_size = output.NumElements();
+  const size_t lhs_rank = lhs.Rank();
+  const size_t rhs_rank = rhs.Rank();
+  const size_t output_rank = output.Rank();
+
+  const int32_t* lhs_batching_dimensions_data =
+      lhs_batching_dimensions.GetDataAs<DataType::kSI32>();
+  const int32_t* rhs_batching_dimensions_data =
+      rhs_batching_dimensions.GetDataAs<DataType::kSI32>();
+  const int32_t* lhs_contracting_dimensions_data =
+      lhs_contracting_dimensions.GetDataAs<DataType::kSI32>();
+  const int32_t* rhs_contracting_dimensions_data =
+      rhs_contracting_dimensions.GetDataAs<DataType::kSI32>();
+  const DimensionSize lhsb_size = lhs_batching_dimensions.NumElements();
+  const DimensionSize lhsc_size = lhs_contracting_dimensions.NumElements();
+  const DimensionSize rhsb_size = rhs_batching_dimensions.NumElements();
+  const DimensionSize rhsc_size = rhs_contracting_dimensions.NumElements();
+
+  // function to generate indices for output
+  auto GenerateIndices = [&](size_t index) -> void {
+    size_t rank = op.output_shape.size();
+    size_t divisor = 1;
+    for (size_t i = 0, j = rank - 1; i < rank; ++i, --j) {
+      op.output_index[j] = (index / divisor) % op.output_shape[j];
+      divisor *= op.output_shape[j];
+    }
+  };
+  // function to incremement lhs and rhs indices
+  auto IncrementIndices = [&]() -> bool {
+    if (lhsc_size == 0) return false;
+    for (DimensionSize i = lhsc_size - 1; i >= 0; --i) {
+      op.lhs_index[lhs_contracting_dimensions_data[i]]++;
+      op.rhs_index[rhs_contracting_dimensions_data[i]]++;
+      if (op.lhs_index[lhs_contracting_dimensions_data[i]] <
+          lhs.shape().Dim(lhs_contracting_dimensions_data[i]))
+        return true;
+      if (i == 0) return false;
+      op.lhs_index[lhs_contracting_dimensions_data[i]] = 0;
+      op.rhs_index[rhs_contracting_dimensions_data[i]] = 0;
+    }
+    return true;
+  };
+  // pre compute helper for lhs and rhs indices
+  DimensionSize lhs_dim_accumulator = 1, rhs_dim_accumulator = 1;
+  for (size_t i = 0; i < lhs_rank; ++i) {
+    lhs_dim_accumulator *= lhs.shape().Dim(i);
+    op.lhs_index_helper[i] = lhs_size / lhs_dim_accumulator;
+  }
+  for (size_t i = 0; i < rhs_rank; ++i) {
+    rhs_dim_accumulator *= rhs.shape().Dim(i);
+    op.rhs_index_helper[i] = rhs_size / rhs_dim_accumulator;
+  }
+
+  StorageT output_element(0);
+  DimensionSize lhs_element_index = 0, rhs_element_index = 0;
+  for (DimensionSize k = 0; k < output_size; ++k, ++output_data) {
+    GenerateIndices(k);
+    std::fill(op.lhs_index.begin(), op.lhs_index.end(), 0);
```

Comment:
`absl::c_fill` can also work and allows you to get rid of the `begin`/`end` pair.

Reply:
changed to absl::c_fill(op.lhs_index, 0);